src/repositories/indexable-repository.php

```diff
@@ -321,13 +332,23 @@ public function find_by_multiple_ids_and_type( $object_ids, $object_type, $auto_
 	 * @return Indexable[] All ancestors of the given indexable.
 	 */
 	public function get_ancestors( Indexable $indexable ) {
-		$hierarchy_table = Yoast_Model::get_table_name( 'Indexable_Hierarchy' );
+		// In the get_ancestors function first do a query purely on the indexable hierarchy table.
+		$ancestors = $this->hierarchy_repository->get_ancestors( $indexable );
+
+		if ( empty( $ancestors ) ) {
+			return [];
+		}
+
+		$indexables = array_column( $ancestors, 'ancestor_id' );
+
+		// If this query returns a single result with an ancestor_id of -1 then return an empty array.
+		if ( $indexables[0] === 0 && count( $indexables ) === 1 ) {
```

Comment:
Should this be `=== -1`? As described in the comment.
```suggestion
		if ( $indexables[0] === -1 && \count( $indexables ) === 1 ) {
```

Reply:
Removed the comments, I've added these to write the code.